kotlin/vector-tiled-layer-custom-style/src/main/java/com/esri/arcgisruntime/sample/vectortiledlayercustomstyle/MainActivity.kt

```diff
@@ -0,0 +1,293 @@
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.BaseAdapter
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
+import com.esri.arcgisruntime.concurrent.Job
+import com.esri.arcgisruntime.data.VectorTileCache
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.layers.ArcGISVectorTiledLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.ItemResourceCache
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.tasks.vectortilecache.ExportVectorTilesTask
+import com.esri.arcgisruntime.sample.vectortiledlayercustomstyle.databinding.ActivityMainBinding
+import kotlinx.android.synthetic.main.activity_main.*
+
+//[DocRef: END]
+
+class MainActivity : AppCompatActivity() {
+
+    // A list of portal item IDs for the online layers.
+    private var onlineItemIds: Array<String> = arrayOf(
+        "1349bfa0ed08485d8a92c442a3850b06",
+        "bd8ac41667014d98b933e97713ba8377",
+        "02f85ec376084c508b9c8e5a311724fa",
+        "1bf0cc4a4380468fbbff107e100f65a5")
+
+    // A list of portal item IDs for the layers which custom style is applied from local resources.
+    private var offlineItemIds: Array<String> = arrayOf(
+        // A vector tiled layer created by the local VTPK and light custom style.
+        "e01262ef2a4f4d91897d9bbd3a9b1075",
+        // A vector tiled layer created by the local VTPK and dark custom style.
+        "ce8a34e5d4ca4fa193a097511daa8855")
+
+    // The item ID of the currently showing layer.
+    private var currentItemID: String = onlineItemIds[0]
+
+    // A dictionary to cache loaded vector tiled layers.
+    private var vectorTiledLayersMap: MutableMap<String, ArcGISVectorTiledLayer> = mutableMapOf()
+
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // Authentication with an API key or named user is required to access basemaps and other location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // Sets up the spinner to change the selected Vector Styled Layer
+        setUpSpinner()
+
+        // Set the currentItemID to default layer.
+        currentItemID = onlineItemIds[0]
+    }
```

Comment:
create a map online once in onCreate 

`        mapView.map = ArcGISMap()`

and remove it from the setMap(...) function